src/commands/synthetics/utils/public.ts

```diff
@@ -726,31 +727,8 @@ export const getTestsToTrigger = async (
     )
   )
 
-  // Keep track of uploaded applications to avoid uploading them twice.
-  const uploadedApplicationByPath: {[applicationFilePath: string]: {applicationId: string; fileName: string}[]} = {}
-
-  for (const item of testsAndConfigsOverride) {
-    // Ignore not found and skipped tests.
-    if ('errorMessage' in item || !('test' in item)) {
-      continue
-    }
-
-    const {test, overriddenConfig} = item
-
-    if (test.type === 'mobile') {
-      const {config: userConfigOverride} = triggerConfigs.find(({id}) => id === test.public_id)!
-      try {
-        await uploadApplicationAndOverrideConfig(
-          api,
-          test,
-          userConfigOverride,
-          overriddenConfig,
-          uploadedApplicationByPath
-        )
-      } catch (e) {
-        throw new CriticalError('UPLOAD_MOBILE_APPLICATION_TESTS_FAILED', e.message)
-      }
-    }
+  if (testsAndConfigsOverride.some((item) => 'test' in item && item.test.type === 'mobile')) {
```

Comment:
That's not possible because `item` is a type union (`TestNotFound | TestSkipped | TestWithOverride`), and the `test` property is only there in 1 case. :grin: I think the current code is ok 👌

Reply:
But that's exactly the point of the `?.` operator -- so you can concisely dereference possibly non-existent fields.

[Here's a simple example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgIwGYEsIBsAmUC8UDeAUFCVAMYD2F6AXFAHZwC2CEAToQL6GiRQCCuAsVL0IAQ2B1GLdgBoRJJiABiaLHXjJ1mBd17QAQkKKkGEqQ2as2enuGgBhE4qgB3Cm+nX5XKPb4AFQgAZ2AhQQAfKGNox0JKejCoS2DkvHwxSToAVjkoZTUMTDp8Smo6AEZObkSQ6ggAOnQKAHMACmAAfkbCnR7y9ABKIA)

```
type Subfield = {
    cool: number
}
type A = {
    neat: number,
    myField: Subfield,
}
type B = {
    neat: number,
}
type C = {
    wow: number,
} 
type Test = A | B | C
const t: Test = {neat: 5, myField: {cool: 1}}
console.log(t?.myField?.cool)
```

Reply:
Oh no! You're right 😆 -- I unintentionally manipulated the type system. This fails once you wipe out the fact that `t` is an `A`

https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgIwGYEsIBsAmUC8UDeAUFCVAMYD2F6AXFAHZwC2CEAToQL6GiRQCCuAsVL0IAQ2B1GLdgBoRJJiABiaLHXjJ1mBd17QAQkKKkGEqQ2as2enuGgBhE4qgB3Cm+nX5XKPb4AFQgAZ2AhQQAfKGNox0JKejCoS2DkvHwxSToAVjkoZTUMTDp8Smo6AEZOTihxELr6EASKJOoIADp0CgBzAApgDsKdAH4O8vQASiA

Reply:
No this won't work 😄. Your example only works because TypeScript inferred `t` as `A`:

[Your example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgIwGYEsIBsAmUC8UDeAUFCVAMYD2F6AXFAHZwC2CEAToQL6GiRQCCuAsVL0IAQ2B1GLdgBoRJJiABiaLHXjJ1mBd17QAQkKKkGEqQ2as2enuGgBhE4qgB3Cm+nX5XKPb4AFQgAZ2AhQQAfKGNox0JKejCoS2DkvHwxSToAVjkoZTUMTDp8Smo6AEZObkSQ6ggAOnQKAHMACmAAfkbCnR7y9ABKQgB6UbNJqAA9LqA):

![image](https://github.com/DataDog/datadog-ci/assets/9317502/6e75925c-9aed-4979-85ea-e52cfa5d3dc0)

[versus when using `as Test`](https://www.typescriptlang.org/play?#code/C4TwDgpgBAygrgIwGYEsIBsAmUC8UDeAUFCVAMYD2F6AXFAHZwC2CEAToQL6GiRQCCuAsVL0IAQ2B1GLdgBoRJJiABiaLHXjJ1mBd17QAQkKKkGEqQ2as2enuGgBhE4qgB3Cm+nX5XKPb4AFQgAZ2AhQQAfKGNox0JKejCocLx8MUk6AFY5KGU1DEw6fEpqOgBGTk4ocRCoYLCEiiTqCAA6dAoAcwAKYAB+NvydQdL0AEpCAHopszmoAD1+oA), to prevent TypeScript from inferring `t` as `A`:

![image](https://github.com/DataDog/datadog-ci/assets/9317502/14d32278-6bcc-4cdc-8ef3-1408a997d072)

Now you get the error which @dd-spells was having (legitimately):

```ts
Property 'myField' does not exist on type 'Test'.
  Property 'myField' does not exist on type 'B'.(2339)
```